src/ts-mockito.ts

```
@@ -32,7 +32,18 @@ export function spy<T>(instanceToSpy: T): T {
 }
 
 export function mock<T>(clazz: { new(...args: any[]): T; } | (Function & { prototype: T }) ): T {
-    return new Mocker(clazz).getMock();
+    return new Mocker(clazz, false).getMock();
+}
+
+export function imock<T>(): T {
+    class Empty {}
+    const mockedValue = new Mocker(Empty, true).getMock();
+
+    if (typeof Proxy === "undefined") {
+        return mockedValue;
```

I think that we should throw Error here to get know user that interface mocking is not available without Proxy support. What do you think?

---

Yes, you are right